ext/msgpack/packer.c

```diff
@@ -124,8 +124,18 @@ void msgpack_packer_write_hash_value(msgpack_packer_t* pk, VALUE v)
 void msgpack_packer_write_other_value(msgpack_packer_t* pk, VALUE v)
 {
     int ext_type;
-    VALUE proc = msgpack_packer_ext_registry_lookup(&pk->ext_registry,
-            rb_obj_class(v), &ext_type);
+
+    VALUE lookup_class;
+
+    if (SYMBOL_P(v)) { /* Symbols have no singleton class */
```

Comment:
This code checks only for Symbol objects, but there are some more classes to raise TypeError for `#singleton_class` method (e.g., Integer, Float).
Is it intended not to check these classes? I know that these classes should be packed in other functions, but i'm not sure there are no other classes which raises TypeError for `#singleton_class`.